src/app/components/layers-panel/layers-panel.component.html

```diff
@@ -1,12 +1,99 @@
+            <div #gripHandle
+                 cdkDragHandle
+                 class="tile-server-move">
+              <i class="fas fa-grip-vertical"></i>
+            </div>
+            <label class="tile-server-visibility">
+              <input name=""
+                     type="checkbox"
+                     [checked]="ts.uiOptions.isActive"
+                     (change)="toggleTileServer(ts)"/>
+              <i class="fas fa-eye-slash unchecked"></i>
+              <i class="fas fa-eye checked"></i>
+            </label>
+            <div class="tile-server-name-container">
+              <input #activeText
+                     id=property-input
+                     type="text"
+                     *ngIf="ts.uiOptions.showInput"
+                     (blur)="updateName(activeText.value, ts)"
+                     (keyup.enter)="updateName(activeText.value, ts)"
+                     [value]="ts.name">
+              <div id="tile-server-name"
+                   class="tile-server-name"
+                   data-tooltip
+                   tabindex="1"
+                   title="{{ts.name}}"
+                   data-position="top"
+                   data-alignment="left"
+                   *ngIf="!ts.uiOptions.showInput"
+                   (dblclick)="showInput(ts, true)">
+                {{ts.name}}
+              </div>
+            </div>
+          </div>
+
+          <div class="cell small-4 grid-x tile-server-right">
+            <button class="button secondary clear edit-buttons">
+              <i class="fas fa-edit edit-button"
+                 (click)="showInput(ts, true)"
+                 *ngIf="!ts.uiOptions.showInput"></i>
+              <i class="fas fa-check edit-button"
+                 (click)="showInput(ts, false)"
+                 *ngIf="ts.uiOptions.showInput"></i>
+            </button>
+            <button class="button secondary clear option-button"
+                    (click)="toggleDescription(ts)">
+              <i class="fas fa-sliders-h" ></i>
+            </button>
+            <button class="button secondary clear button-no-margin delete-button"
+                    (click)="deleteTileServer(ts)">
+              <i class="fa fa-trash"></i>
+            </button>
+          </div>
+        </div>
+        <div *ngIf="ts.uiOptions.showDescription">
+          <div class="grid-x row tile-server-description">
+            <label class="small-4 align-self-middle">Opacity</label>
+            <input #slider type="range" min="0" max="1" step="0.1" [value]="ts.uiOptions.opacity" class="slider small-8 align-self-middle tile-server-slider" id="myRange" (change)="setLayerOpacity(ts, slider.value)">
+          </div>
+        </div>
+      </div>
+    </div>
+
+    <div *ngIf="dirtyOptions" class="tile-server-dirty-options">
+      <i class="tile-server-dirty-options-warning">
+        Layer options have been changed!
+        <br>
+        Save to persist changes.
```

Comment:
Should this "concept of editing but not persisting" be extended to include layers as well.  So that any changes to layers (new or delete) does not take effect unless the user saves.

Not certain what to do here regarding adding/deleting layers but wanted to discuss.

Reply:
@nathanfranklin At first, I tried doing that and ran into some UX problems (I can't remember specifically).

Reply:
This is being put in a ticket.